docs/DeveloperGuide.md

```diff
@@ -193,6 +193,97 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+Once logged in, the user will have access to all the commands available.
+
+The format for the `login` command can be found [here](UserGuide.md#login).
+
+**Feature details:**
+
+1. The user specifies the `Username` and `Password` in the `login` command.
+2. If any of the fields is not provided, an error message with the correct command usage will be shown.
+3. If invalid command parameters are provided, an error message with the correct parameter format will be shown.
+4. The `User` is then cross-referenced with the stored user in `Model` to check if the credentials match.
+If incorrect credentials are provided, an error message regarding wrong credentials will be shown.
+5. If the user is currently logged in, an error message will be shown.
+6. If all the previous steps are completed without exceptions, the user will be logged in and the 
+`isLoggedIn` status in `Model` will be updated.
+
+The following activity diagram shows the logic of a user logging in:
+
+<puml src="diagrams/UserLoginActivityDiagram.puml" alt="UserLoginActivityDiagram" />
+
+The sequence of the `login` command is as follows:
+
+1. Upon launching the application, the `ModelManager` will be initialized with 
+the `User` constructed with details from the authentication.json file.
+2. The user inputs the `login` command with the username and password.
+3. The `userLoginCommandParser` checks whether all the required fields are present.
+If all fields are present, it creates a new `userLoginCommand`.
+4. The `userLoginCommand` checks whether the user is currently logged in by calling `Model#getUserLoginStatus()`. 
+5. The `userLoginCommand` then checks if the user credentials match the stored user by calling `Model#userMatches()`.
+6. If the user is not logged in and the credentials match, the `userLoginCommand` calls `Model#setLoginSuccess()`,
+changing the login status to true and enabling the user access to all commands.
+7. The `userLoginCommand` also calls `Model#updateFilteredPersonList()` to display the list of customers.
+
+The following sequence diagram shows how the `login` command works:
+
+<puml src="diagrams/UserLoginSequenceDiagram.puml" alt="UserLoginSequenceDiagram" />
+
+### User Logout Command
+
+**Overview:**
+
+The `logout` command is used to log out from the user's account.
+Once logged out, the user will have no access to all the commands available, except for `help`, `exit`, 
+`register` and `delete account`.
+
+The format for the `logout` command can be found [here](UserGuide.md#logout).
+
+**Feature details:**
+
+1. The user executes the `logout` command.
+2. If extra command parameters are provided after specifying `logout`, the logout command will still be executed.
+3. If the user is currently logged out, an error message will be shown.
+4. If all the previous steps are completed without exceptions, the user will be logged out and the 
+`isLoggedIn` status in `Model` will be updated to `false`.
+
```

Comment:
Would it be better if the code corresponding step in the `login` command also indicates that the status is updated to `true`

Reply:
Updated the `login` command section